### PR TITLE
RHCOS4 STIG: Cover controls that correspond to NIST AC

### PR DESCRIPTION
**Pull request by jhrozek**
#### Description:

- OCP4 STIG: SRG-APP-000343-CTR-000780 and SRG-APP-000381-CTR-000905 are covered by `audit_rules_suid_privilege_function` - this one is quite easy
- SRG-APP-000029-CTR-000085: Audit execution of all setuid and setgid binaries on RHCOS4 - this one is more tricky. First, there's quite a few new rules that were added because I didn't find rules equivalent to the check and fix in the STIG draft. Second, I'm not sure if this SRG is the right one to be covered by the rules - the rules are valid and should be included in the STIG, but I think that SRG-APP-000381-CTR-000905 would be better suited for these checks. But at this point I'm unsure what to do and I think we should start an epic with stories that tracks the next STIG update to shuffle things around. Finally, I think some of the rules are following the letter of "let's audit all SUID binaries", but are not really needed, e.g. the SSSD binaries are really just helpers that can't be executed by a user and the grub2 binary probably shouldn't be on RHCOS in the first place (there's no GRUB on RHCOS...)

#### Rationale:

- RHCOS4 STIG

#### Review Hints:

- Make sure that the e2e tests pass
- Make sure that the checks and remediations are in line with the AC controls in the [STIG draft](https://docs.google.com/spreadsheets/d/19XSsjNZz5KJ4o7dEboOscyg_YJM_ZH7I_W72O9t1Cig/edit#gid=676941970). I recommend making a copy of the document, then filtering column A with any controls that start with `AC` and looking for `MachineConfig` in column I (Fix).

### CODE REVIEW

**Comment by jhrozek** [comment]
@ggbecker @yuumasato I'm at loss to debug the CI issues:
```
ValueError: The rule 'audit_rules_privileged_commands_dbus_daemon_launch_helper' isn't mapped to any component! Insert the rule ID at least once to the rule-component mapping.
```
what is a rule-component mapping?

and then this test fails:
```
./shared/templates/audit_rules_privileged_commands/tests/auditctl_missing_auid.fail.sh
```
why does it fail? What does the test do?

**Comment by yuumasato** [comment]
@jhrozek I'm not sure, bu I **think** you need to add the new rules into any fine in  `components/`.
`components/audit.yml` seems to be the most fit.

CC @matejak and @jan-cerny

**Comment by jhrozek** [comment]
> @jhrozek I'm not sure, but I **think** you need to add the new rules into any file in `components/`. `components/audit.yml` seems to be the most fit.
> 
> CC @matejak and @jan-cerny

thanks, let's see how the new CI run goes

**Comment by jhrozek** [comment]
RHCOS STIG CI succeeded.
@Vincent056 @rhmdnd PTAL

**Comment by jhrozek** [comment]
direct link to e2e test log: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/ComplianceAsCode_content/10727/pull-ci-ComplianceAsCode-content-master-e2e-aws-ocp4-stig/1671091507496488960/artifacts/e2e-aws-ocp4-stig/test/build-log.txt

**Comment by jhrozek** [comment]
Correct test log: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/ComplianceAsCode_content/10727/pull-ci-ComplianceAsCode-content-master-e2e-aws-rhcos4-stig/1671091479709224960/artifacts/e2e-aws-rhcos4-stig/test/build-log.txt

File: controls/srg_ctr/SRG-APP-000029-CTR-000085.yml [hunk 1]
```diff
@@ -7,9 +7,27 @@ controls:
...
+  - audit_rules_privileged_commands_sssd_krb5_child
+  - audit_rules_privileged_commands_sssd_ldap_child
+  - audit_rules_privileged_commands_sssd_proxy_child
+  - audit_rules_privileged_commands_sssd_selinux_child
+  - audit_rules_privileged_commands_su
+  - audit_rules_privileged_commands_sudo
+  - audit_rules_privileged_commands_umount
+  - audit_rules_privileged_commands_polkit_helper
+  - audit_rules_privileged_commands_pam_timestamp_check
+  - audit_rules_privileged_commands_unix_chkpwd
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply controls/srg_ctr/SRG-APP-000029-CTR-000085.yml:31]: no, it is an oversight, let me take a look. Thanks!

[the remark this answers is not in the document]
Comment (jhrozek) [reply controls/srg_ctr/SRG-APP-000029-CTR-000085.yml:31]: just as an explanation - those binaries are setgid, not setuid, maybe that's why I missed them. Nonetheless, the rules must be added - thanks!

[the remark this answers is not in the document]
Comment (jhrozek) [reply controls/srg_ctr/SRG-APP-000029-CTR-000085.yml:33]: the latest patch should add those watches, let's try the CI run...

**Comment by jhrozek** [comment]
lol oops I really should get into the habit of running some local checks :-)

**Comment by jhrozek** [comment]
Some CCEs were not unique, hopefully now all CI runs will be green

File: controls/srg_ctr/SRG-APP-000029-CTR-000085.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuumasato) [reply controls/srg_ctr/SRG-APP-000029-CTR-000085.yml:33]: Looks good, the only thing I miss is the `srg:` in `audit_rules_privileged_commands_ssh_keysign`.

We can wait for e2e test to finish, then add srg and merge.

**Comment by yuumasato** [comment]
@jhrozek e2e passed.
Can you just add `srg` to `audit_rules_privileged_commands_ssh_keysign`, then I we can merge.

**Comment by jhrozek** [comment]
> @jhrozek e2e passed. Can you just add `srg` to `audit_rules_privileged_commands_ssh_keysign`, then I think we can merge.

nice catch, done

**Review by yuumasato — APPROVED** [review]

**Comment by yuumasato** [comment]
The Automatus CS8 and CS9 tests fail because some rules are only available for rhcos. Automatus Fedora is failing to prepare environment for tests, for some reason.
The rpm-build on Fedora is failing due to `import imp`.

**Pull request merged**